torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1099,6 +1117,329 @@ def call_function(
+        "SEND",
+        "MAKE_FUNCTION",
+        # This could be supported in the future.
+        "LOAD_DEREF",
+        # Exceptions
+        "RAISE_VARARGS",
+        "SETUP_FINALLY",
+        "POP_EXCEPT",
+        # TODO: Nested loops are not supported for now.
+        # We can support them by recursively calling this translation every time
+        # for each nested loop, from the innermost loop to the outermost.
+        #
+        # E.g.
+        # loop1:
+        #   loop2:
+        #     loop3:
+        #
+        # Compile loop3, then loop2, then loop1
+        "FOR_ITER",
+        # These two are fine for the end of the loop body but not within.
+        # TODO: we can support these too. They just need to be transformed to the
+        # right RETURN_VALUEs (see code below for how to do that).
+        "JUMP_BACKWARD",
+        "JUMP_ABSOLUTE",
+        # list, set, dictionary comprehensions
+        "LIST_APPEND",
+        "SET_ADD",
+        "MAP_ADD",
+    }
+
+    store_target: int
+
+    NOT_SET_SENTINEL = 0xDEADBEEF
+
+    def __init__(self, value: "RangeIteratorVariable", **kwargs):
+        from ..source import GlobalSource
+
+        super().__init__(value, source=GlobalSource("__builtins__.range"), **kwargs)
+        self.store_target = -1
+
+    @staticmethod
+    def make_self(
+        tx,
+        value: "RangeIteratorVariable",
+        real_globals: dict,
+        # The code object the loop was defined in.
+        host_code_object: types.CodeType,
+        loop_body_instructions: List["Instruction"],
+        symbolic_locals: Dict[str, VariableTracker],
+    ):
+        from . import ConstantVariable
+
+        if (
+            loop_items := len(value.items)
+        ) < torch._dynamo.config.for_loop_medium_size_boundary:
+            raise CannotConvertRangeToHigherOrder(
+                f"Loop of length {loop_items} too small to consider optimizing"
+            )
+        assert loop_body_instructions[-1].opname in {"JUMP_BACKWARD", "JUMP_ABSOLUTE"}
+        if loop_body_instructions[0].opname == "UNPACK_SEQUENCE":
+            raise CannotConvertRangeToHigherOrder(
+                "Unpacking a range iterator, seems wrong."
+            )
+        if any(
+            op.opname in RangeHigherOrderVariable.FORBIDDEN_OPCODES
+            for op in loop_body_instructions[:-1]
+        ):
+            raise CannotConvertRangeToHigherOrder(
+                "Control flow too complex for loop higher order op."
+            )
+        varnames = host_code_object.co_varnames
+        args = [
+            symbolic_locals.get(k)
+            or ConstantVariable.create(RangeHigherOrderVariable.NOT_SET_SENTINEL)
+            for k in varnames
+        ]
+        # STORE_FAST always follows a FOR_ITER as CPython needs to store the next(iter) into the local
+        # E.g. in `for i in range(10)`, there is a `STORE_FAST i``
+        assert loop_body_instructions[0].opname == "STORE_FAST"
+        co_code: List[int] = []
```

Comment:
Any particular reason why we're avoiding `torch._dynamo.bytecode_transformation` in order to create the new code object? It provides some nice utilities such as dealing with CACHE entries and utility functions to create instructions. 

e.g. https://github.com/pytorch/pytorch/blob/4ceb44c40d52c57143ab1dbc8c14373f5205e9fb/torch/_dynamo/resume_execution.py#L492

Reply:
Didn't know of that, will use it !